### PR TITLE
[SOLR-15377] Do not swallow execption stack traces in IndexFetcher

### PR DESCRIPTION
**Pull request by tboeghk**
# Description

The `IndexFetcher` swallows Exceptions upon replication errors and does not log the stack trace properly. This led to some confusion when setting up a `TLOG/PULL` replica ensemble.

# Solution

This PR adds the exception's stack trace into the log messages upon replication abort. This will help users for further analysis.

# Tests

No tests needed (I guess)

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/handler/IndexFetcher.java
Comment (cpoerschke) [inline solr/core/src/java/org/apache/solr/handler/IndexFetcher.java:685]: Locally this does not compile for me on Java 11, maybe instead this?
```suggestion
        throw (InterruptedException)(new InterruptedException("Index fetch interrupted").initCause(e));
```

Comment (tboeghk) [reply solr/core/src/java/org/apache/solr/handler/IndexFetcher.java:685]: You're absolutely right! Way better fix!

File: solr/core/src/java/org/apache/solr/handler/IndexFetcher.java
Comment (cpoerschke) [inline solr/core/src/java/org/apache/solr/handler/IndexFetcher.java:685]: observation: changing the details of the exception being thrown here (not its type but it having or not having a cause or a particular type of cause) could theoretically change the calling code's behaviour. practically that seems unlikely though. also inspection of calling code finds no obvious signs that behaviour would change.

**Review by cpoerschke — APPROVED** [review]
LGTM

**Comment by cpoerschke** [comment]
Hmm, I just tried to merge in the latest `origin/main` and add the `solr/CHANGES.txt` entry before merging here but then github wouldn't let me push the commits to the pull request branch.

@tboeghk would you be comfortable enabling maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for this PR branch? Or could I ask you to add the `solr/CHANGES.txt` entry e.g. something like

```
* SOLR-15377: Improve IndexFetcher's exception logging upon replication errors. (Torsten Bøgh Köster, Christine Poerschke)
```

in (say) the "Other Changes" sub-section of the  8.9.0 section? Thanks!

**Comment by tboeghk** [comment]
Unfortunately this  PR originated from an organizational fork and not from a personal repo, so I cannot enable access for maintainers ([bug in Github](https://github.com/isaacs/github/issues/1681)). 

But I happily added the change to `solr/CHANGES.txt`!

**Comment by cpoerschke** [comment]
> Unfortunately this PR originated from an organizational fork and not from a personal repo, so I cannot enable access for maintainers ([bug in Github](https://github.com/isaacs/github/issues/1681)).

Ah, interesting, thanks for sharing!

> But I happily added the change to `solr/CHANGES.txt`!

Thank you.

**Pull request merged**
